### PR TITLE
Add documentation for `kedro run` lifecycle

### PR DESCRIPTION
**Pull request by noklam**
## Description

Add a diagram to explain the hook execution order and rearrange the hooks spec in docs to match the execution order.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: docs/source/hooks/introduction.md [hunk 1]
```diff
@@ -32,6 +32,8 @@ The naming convention for error hooks is `on_<noun>_error`, in which:
 
 [kedro.framework.hooks](/kedro.framework.hooks) lists the full specifications for which you can inject additional behaviours by providing an implementation.
 
+This diagram illustrate the execution order of hooks during `kedro run`:
```
Comment (stichbury) [inline docs/source/hooks/introduction.md:35]:
```suggestion
This diagram illustrates the execution order of hooks during `kedro run`:
```

**Review by stichbury — APPROVED** [review]
This looks great to me, thanks! If you check the page, the background to the image is grey -- did you intend this? I really liked the use of the Kedro colour: nice attention to detail!

**Review by ankatiyar — APPROVED** [review]
Thanks @noklam!

**Comment by noklam** [comment]
I have update the image to a white background.

File: docs/source/hooks/introduction.md [hunk 2]
```diff
@@ -32,6 +32,8 @@ The naming convention for error hooks is `on_<noun>_error`, in which:
 
 [kedro.framework.hooks](/kedro.framework.hooks) lists the full specifications for which you can inject additional behaviours by providing an implementation.
 
+This diagram illustrates the execution order of hooks during `kedro run`:
+![kedro run hook execution order](../../source/meta/images/kedro_run_lifecycle.png)
```
Comment (astrojuanlu) [inline docs/source/hooks/introduction.md:36]: Can we remove the reference to `source`?

```suggestion
![kedro run hook execution order](../meta/images/kedro_run_lifecycle.png)
```

Comment (astrojuanlu) [reply docs/source/hooks/introduction.md:36]: Also @noklam if you did this with Mermaid, can you paste the code in a comment? Similar to how we're doing it in other diagrams, see for example 

https://github.com/kedro-org/kedro/blob/d433ff31542ee03724930e2f409e9fd5d9819a1e/docs/source/get_started/new_project.md?plain=1#L170-L187

**Review by astrojuanlu — COMMENTED** [review]
I love having a diagram, finally! Added a couple of comments.

Other than that, in terms of structure I find this navigation structure a bit weird:

image

And also the fact that the page is titled "Hooks" but the URL is "introduction" could have SEO implications @stichbury?

(I know this problem was not introduced by this PR, but since I looked at the overall result, I wanted to flag it anyway)

**Pull request merged**